unlock-app/src/hooks/useCreditCardEnabled.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from '@tanstack/react-query'
+import { storage } from '~/config/storage'
+
+interface CreditCardEnabledProps {
+  lockAddress: string
+  network: number
+}
+
+export const useCreditCardEnabled = ({
+  lockAddress,
+  network,
+}: CreditCardEnabledProps) => {
+  return useQuery(['useCreditCardEnabled'], async () => {
```

Comment:
You need to pass lockAddress and network as params to query so it doesn't cache it for all locks